### ISSUE TITLE
Ensuring modules can be rebuilt

### ISSUE THREAD

**Issue by ocaisa** [EESSI/software-layer#101]
When including software in the EESSI stack, we should not only ensure that the software can be built on the various archs, but also that `--module-only` returns an identical set of modules as the original installation.

**Comment by ocaisa** [comment]
This will also allow us to easily perform the sanity checks on different OSes (and help us to avoid issues like #99 )

**Comment by ocaisa** [comment]
This is something that we could do in our CI since rebuilding the module tree doesn't take *too* long.

**Comment by ocaisa** [comment]
`--module-only --rebuild` will do that for us and will also compare module files...but because of `--robot` it won't check any dependencies

**Comment by Flamefire** [comment]
True. Supporting module-only rebuilds would still be good as some modules encode config options such as PYTHONPATH vs EBPYTHONPATHS and CPATH

However this is hard for 3 reasons I encountered:
- An existing easyconfig (from the reprod folder) generally requires a specific version of EasyBuild or at least easyblocks as behavior might be changed and corresponding easyconfig parameters might be added/changed/removed
- Rebuilding using the easyconfig from another EasyBuild version might fail sanity checks, e.g. Abseil switched from dynamic to static libs and the sanity check expects either or
- Using any EasyBuild version might not work with existing config files/env-variables as EasyBuild options are added/removed/changed and an error will be issues on those. Or defaults might have changed.

So shipping both module trees would indeed help and having reliable module-only rebuilds (in the same configuration/versions at least) is required for that anyway.
